src/gpu/GpuSieve.cu

```diff
@@ -0,0 +1,126 @@
+#include <cmath>
+#include <cuda_runtime.h>
+#include "GpuSieve.hpp"
+#include <iostream>
+#include <vector>
+#include <cassert>
+
+// Maximum stride for GPU processing
+const uint64_t MAX_STRIDE = 256000000;
+// Number of threads per block (adjustable based on GPU)
+int BLOCK_SIZE = 256;
+
+// Constructor: Initialize GpuSieve object with no allocated memory on the host
+GpuSieve::GpuSieve() : maxLimit(0), isPrimeListHost(nullptr) {}
+
+// Destructor: Free host memory if allocated
+GpuSieve::~GpuSieve() {
+    // Free host memory
+    if (isPrimeListHost) {
+        free(isPrimeListHost);
+        isPrimeListHost = nullptr;
+    }
```

Comment:
leave the constructor empty as unique pointer handles memory dealocation automatically